diesel_codegen/src/schema_inference/mod.rs

```diff
@@ -73,25 +76,16 @@ pub fn infer_schema_body<T: Iterator<Item=P<ast::Expr>>>(
     Ok(MacEager::items(SmallVector::many(try!(impls))))
 }
 
-fn establish_connection(
-    cx: &mut ExtCtxt,
-    sp: Span,
-    database_url: &str,
-) -> Result<PgConnection, Box<MacResult>> {
-    PgConnection::establish(database_url).map_err(|_| {
-        cx.span_err(sp, "failed to establish a database connection");
-        DummyResult::any(sp)
-    })
-}
+
 
 fn table_macro_call(
     cx: &mut ExtCtxt,
     sp: Span,
-    connection: &PgConnection,
+    connection: &InferConnection,
     table_name: &str,
 ) -> Result<P<ast::Item>, Box<MacResult>> {
     match get_table_data(connection, table_name) {
-        Err(NotFound) => {
+        Err(::diesel::result::Error::NotFound) => {
```

Comment:
Because the compiler is telling me to do this. (Don't know where the other NotFound is coming from)
```
src/schema_inference/mod.rs:93:13: 93:21 warning: pattern binding `NotFound` is named the same as one of the variants of the type `diesel::result::Error` [E0170]
src/schema_inference/mod.rs:93         Err(NotFound) => {
                                           ^~~~~~~~
src/lib.rs:22:1: 22:23 note: in this expansion of include!
src/schema_inference/mod.rs:93:13: 93:21 help: run `rustc --explain E0170` to see a detailed explanation
src/schema_inference/mod.rs:93:13: 93:21 help: if you meant to match on a variant, consider making the path in the pattern qualified: `diesel::result::Error::NotFound`
src/schema_inference/mod.rs:97:9: 97:15 error: unreachable pattern [E0001]
src/schema_inference/mod.rs:97         Err(_) => {
```